File: Python/53_Maximum_Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        d = float('-inf')
        m = float('-inf')
        for i in range(len(nums)):
            d = max(nums[i], d + nums[i])
            if d > m:
                m = d
        return m

File: C++/53_Maximum_Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int prev = 0;
        int sum = -INT_MAX;
        for (int x : nums){
            prev = max(prev + x, x);
            if (prev > sum)
                sum = prev;
        }
        return sum;
    }
};